src/main/java/com/onarandombox/multiverseinventories/share/Sharables.java

```diff
@@ -582,6 +590,149 @@ public boolean updatePlayer(Player player, PlayerProfile profile) {
+                public boolean updatePlayer(Player player, PlayerProfile profile) {
+                    Map<String, Integer> playerStats = profile.get(GAME_STATISTICS);
+                    for (Statistic stat : Statistic.values()) {
+                        if (stat.getType() == Statistic.Type.UNTYPED) player.setStatistic(stat, 0);
+                    }
+                    if (playerStats == null) {
+                        return false;
+                    }
+                    for (String stringStat : playerStats.keySet()) {
+                        Statistic stat = Statistic.valueOf(stringStat);
+                        if (stat.getType() == Statistic.Type.UNTYPED) player.setStatistic(stat, playerStats.get(stat.name()));
+                    }
+                    return true;
+                }
+            }).defaultSerializer(new ProfileEntry(false, "game_statistics")).altName("game_stats").build();
+
+    /**
+     * Sharing Recipes.
+     */
+    public static final Sharable<Map> RECIPES = new Sharable.Builder<Map>("recipes", Map.class,
+            new SharableHandler<Map>() {
+                @Override
+                public void updateProfile(PlayerProfile profile, Player player) {
+                    Set<NamespacedKey> recipes = new HashSet<>();
+                    Iterator<Recipe> recipeIterator = inventories.getServer().recipeIterator();
+
+                    while (recipeIterator.hasNext()) {
+                        Recipe recipe = recipeIterator.next();
+                        if (recipe instanceof Keyed) {
+                            NamespacedKey key = ((Keyed) recipe).getKey();
+                            if (player.undiscoverRecipe(key)) recipes.add(key);
+                        }
+                    }
+
+                    player.discoverRecipes(recipes);
+
+                    Map<String, List<String>> recipesMap = new HashMap<>();
+                    for (NamespacedKey recipe: recipes) {
+                        recipesMap.putIfAbsent(recipe.getNamespace(), new ArrayList<>());
+                        recipesMap.get(recipe.getNamespace()).add(recipe.getKey());
+                    }
+
+                    profile.set(RECIPES, recipesMap);
+                }
+
+                @Override
+                public boolean updatePlayer(Player player, PlayerProfile profile) {
+                    Map<String, List<String>> recipes = profile.get(RECIPES);
+                    if (recipes == null) recipes = new HashMap<>();
+                    Iterator<Recipe> recipeIterator = inventories.getServer().recipeIterator();
+
+                    while (recipeIterator.hasNext()) {
+                        Recipe recipe = recipeIterator.next();
+                        if (recipe instanceof Keyed) {
+                            NamespacedKey key = ((Keyed) recipe).getKey();
+                            if (recipes.containsKey(key.getNamespace()) && recipes.get(key.getNamespace()).contains(key.getKey())) {
```

Comment:
Would there be a different if we do this instead? Idk I always feel that doing `containsKey` then `get` is doing something twice.
```java
List<String> recipe = recipes.get(key.getNamespace());
if (recipe != null && .contains(key.getKey())) {
// ...
```